src/wp-includes/rest-api/endpoints/class-wp-rest-navigation-fallback-controller.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * WP_REST_Navigation_Fallback_Controller class
+ *
+ * REST Controller to create/fetch a fallback Navigation Menu.
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 6.3.0
+ */
+
+/**
+ * REST Controller to fetch a fallback Navigation Block Menu. If needed it creates one.
+ *
+ * @since 6.3.0.
+ */
+class WP_REST_Navigation_Fallback_Controller extends WP_REST_Controller {
+
+	/**
+	 * The Post Type for the Controller
+	 *
+	 * @since 6.3.0.
+	 * @var string
+	 */
+	private $post_type;
+
+	/**
+	 * Constructs the controller.
+	 *
+	 * @since 6.3.0.
+	 */
+	public function __construct() {
+		$this->namespace = 'wp-block-editor/v1';
+		$this->rest_base = 'navigation-fallback';
+		$this->post_type = 'wp_navigation';
+	}
+
+	/**
+	 * Registers the controllers routes.
+	 *
+	 * @since 6.3.0.
+	 * @return void
+	 */
+	public function register_routes() {
+
+		// Lists a single nav item based on the given id or slug.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::READABLE ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to read fallbacks.
+	 *
+	 * @since 6.3.0.
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+
+		$post_type = get_post_type_object( $this->post_type );
+
+		// Getting fallbacks requires creating and reading `wp_navigation` posts.
+		if ( ! current_user_can( $post_type->cap->create_posts ) || ! current_user_can( 'edit_theme_options' ) || ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error(
+				'rest_cannot_create',
+				__( 'Sorry, you are not allowed to create Navigation Menus as this user.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( 'edit' === $request['context'] && ! current_user_can( $post_type->cap->edit_posts ) ) {
+			return new WP_Error(
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit Navigation Menus as this user.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Gets the most appropriate fallback Navigation Menu.
+	 *
+	 * @since 6.3.0.
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
```

Comment:
We generally separate `since` and `params` with an empty line (see [example](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-functions-class-methods))

However, `param` and `return` don't need an empty line.